webview/src/plots/components/checkpointPlots/checkpointPlotsSlice.ts

```diff
@@ -0,0 +1,73 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit'
+import {
+  CheckpointPlotData,
+  CheckpointPlotsData,
+  DEFAULT_SECTION_COLLAPSED,
+  DEFAULT_SECTION_SIZES,
+  PlotSize,
+  Section
+} from 'dvc/src/plots/webview/contract'
+import { clearData } from '../../actions'
+import { ReducerName } from '../../constants'
+
+type PlotsById = { [key: string]: CheckpointPlotData }
+export interface CheckpointPlotsState extends CheckpointPlotsData {
+  isCollapsed: boolean
+  hasData: boolean
+  plotsIds: string[]
+  plotsById: PlotsById
+}
+
+export const checkpointPlotsInitialState: CheckpointPlotsState = {
+  colors: { domain: [], range: [] },
+  hasData: false,
+  isCollapsed: DEFAULT_SECTION_COLLAPSED[Section.CHECKPOINT_PLOTS],
+  plots: [],
+  plotsById: {},
+  plotsIds: [],
+  selectedMetrics: [],
+  size: DEFAULT_SECTION_SIZES[Section.CHECKPOINT_PLOTS]
+}
+
+export const checkpointPlotsSlice = createSlice({
+  extraReducers: builder => {
+    builder
+      .addCase(clearData, (_, action) => {
+        if (!action.payload || action.payload === ReducerName.checkpoint) {
+          return { ...checkpointPlotsInitialState }
+        }
+      })
+      .addDefaultCase(() => {})
+  },
+  initialState: checkpointPlotsInitialState,
+  name: ReducerName.checkpoint,
+  reducers: {
+    changeSize: (state, action: PayloadAction<PlotSize>) => {
+      state.size = action.payload
+    },
+
+    setCollapsed: (state, action: PayloadAction<boolean>) => {
+      state.isCollapsed = action.payload
+    },
+    update: (state, action: PayloadAction<CheckpointPlotsData>) => {
+      if (action.payload) {
+        state.plots = action.payload.plots
+        state.colors = action.payload.colors
+        state.selectedMetrics = action.payload.selectedMetrics
+        state.size = action.payload.size
+        state.plotsIds = action.payload?.plots?.map(plot => plot.title) || []
+        state.plotsById = {}
```

Comment:
I haven't done so for template plots because there isn't the same filtering being done in `TemplatePlotsWrapper` as there is in `CheckpointPlotsWrapper`. For template plots, `plots` is used in `TemplatePlots` and isn't passed to any other components.